source/agora/common/TransactionPool.d

```diff
@@ -301,6 +301,65 @@ public class TransactionPool
 
         return true;
     }
+
+    /***************************************************************************
+
+        Check if a transaction hash exists in the transaction pool.
+
+        Params:
+            tx = the transaction hash
+
+        Returns:
+            true if the transaction pool has the transaction hash.
+
+    ***************************************************************************/
+
+    public bool hasTransactionHash (const ref Hash tx) @trusted
+    {
+        auto results = this.db.execute("SELECT key FROM tx_pool " ~
+            "WHERE key = ?", tx[]);
+
+        return !results.empty;
+    }
+}
+
+/// hasTransactionHash tests
+unittest
+{
+    import agora.consensus.Genesis;
+
+    auto pool = new TransactionPool(":memory:");
+    scope(exit) pool.shutdown();  // note: must call outside destructor
+
+    auto gen_key = getGenesisKeyPair();
+    auto txs = makeChainedTransactions(gen_key, null, 1);
+
+    txs.each!(tx => pool.add(tx));
+    assert(pool.length == txs.length);
+
+    foreach (Transaction tx; txs)
```

Comment:
```suggestion
    foreach (tx; txs)
```
I have not seen the use of type elsewhere. Most of them were omitted.

Reply:
It's simple.